### PR TITLE
new pkg: nushell

### LINKED ISSUE #33708: [Wolfi Package Request]: nushell
### What software would you like us to add to wolfi-os. Ideally include a URL to the project and its source.

https://github.com/nushell/nushell
https://pkgs.alpinelinux.org/package/edge/community/x86_64/nushell

### which versions of the software should we include?

latest release: 0.99.1

### Add some justification for why this specific package and versions are important.

This shell is becoming more popular and it would be useful to include. zsh and fish are included, so why not nushell?

### Please check all that apply

- [X] This package has an un-restrictive license
- [X] The package/versions proposed are actively maintained upstream
- [ ] I am interested in adding this package to Wolfi OS myself
- [ ] I am willing to help maintain this package

### PR DESCRIPTION
**Pull request by tuananh**
Fixes: https://github.com/wolfi-dev/os/issues/33708

Related:

#### For new package PRs only

- [x] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates
- [ ] This PR links to the upstream project's support policy (e.g. `endoflife.date`)

#### For new version streams

- [ ] The upstream project actually supports multiple concurrent versions.
- [ ] Any subpackages include the version string in their package name (e.g. `name: ${{package.name}}-compat`)
- [ ] The package (and subpackages) `provides:` logical unversioned forms of the package (e.g. `nodejs`, `nodejs-lts`)
- [ ] If non-streamed package names no longer built, open PR to withdraw them (see [WITHDRAWING PACKAGES](https://github.com/wolfi-dev/os/blob/main/WITHDRAWING_PACKAGES.md))

#### For package updates (renames) in the base images

When updating packages part of base images (i.e. cgr.dev/chainguard/wolfi-base or ghcr.io/wolfi-dev/sdk)
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk images successfully build
- [ ] REQUIRED cgr.dev/chainguard/wolfi-base and ghcr.io/wolfi-dev/sdk contain no obsolete (no longer built) packages
- [ ] Upon launch, does `apk upgrade --latest` successfully upgrades packages or performs no actions

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: nushell.yaml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      mkdir -p ~/.cargo
+      cat >> ~/.cargo/config.toml <<-EOF
+
+        [target.$TARGET]
+        git2 = { rustc-link-lib = ["git2"] }
+        rusqlite = { rustc-link-lib = ["sqlite3"] }
+      EOF
+
+  - runs: |
+      cargo auditable build --workspace --locked --release
```
Comment (kranurag7) [inline nushell.yaml:46]: we can consider using `cargo/build` pipeline [here](https://github.com/chainguard-dev/melange/blob/main/pkg/build/pipelines/cargo/build.yaml)

Comment (tuananh) [reply nushell.yaml:46]: i couldn't use that. in the `target/release`, there are many files but i only want to install some files (executable).

`cargo/build` support either 1 file or all files. 

That's why I still use `uses` pipeline for now.

**Comment by tuananh** [comment]
also, cant bump the deps that causes this `GHSA-2326-pfpj-vx3h fixed in 1.0.0` because cargo.lock specified only `^0.8` and the maintainer jump from `0.8.5` (the version being used) directly to `1.0.0` release.

File: nushell.yaml (hunk 1 above)
Comment (tuananh) [reply nushell.yaml:46]: I think it does support multiple files if i set `output` like `nu nu_plugin_inc foo bar` but then, i will have to list all the plugin names there.

do you prefer it that way?

```yaml
- uses: cargo/build
  with:
    opts: --workspace --release --locked
    output: nu nu_plugin_custom_values nu_plugin_example nu_plugins_formats ....
```

Comment (kranurag7) [reply nushell.yaml:46]: I think range is not supported as of now. 

ref: https://github.com/chainguard-dev/melange/blob/c072b034edbb302e60599c6789630a3f3153c867/pkg/build/pipelines/cargo/build.yaml#L42

**Review by ajayk — APPROVED** [review]

**Pull request merged**